lib/src/model.dart

```diff
@@ -4596,6 +4888,32 @@ class PackageGraph {
     return hrefMap;
   }
 
+  void _collectLibraryNames(library) {
+    void collectClassNames(Class c) {
+      nameRegistry.add(c.package);
+      if (!c._mixins.isEmpty) {
+        c._mixins.forEach((t) {
+          c.nameRegistry.add(t.element);
+        });
+      }
+      if (c._supertype != null) {
+        c.nameRegistry.add(c._supertype.element);
+      }
+      if (!c.interfaces.isEmpty) {
+        c.interfaces.forEach((t) {
+          c.nameRegistry.add(t.element);
+        });
+      }
+    }
+
+    nameRegistry.add(library);
+    library._allModelElements
+        .forEach((ModelElement e) => nameRegistry.add(e.library));
+    library._allModelElements
+        .forEach((ModelElement e) => library.nameRegistry.add(e));
+    library._allClasses?.forEach(collectClassNames);
```

Comment:
What motivated this?  IIUC, interfaces/mixins/supertypes and so on should be picked up by the libraries that contain them, and also not be relevant to the library containing the class they reference.

Reply:
Huh.  Yeah, I guess you're right.  I had that in there before I was making sure to add all the names from the model elements in the library, so maybe that was just redundant.